### DISCUSSION TITLE
index.php ENVIRONMENT not able to be set from environment variable

### DISCUSSION THREAD

**Discussion by T012m3n7oR** [magicbug/Cloudlog#1692] [Ideas / Feature Requests]
I see that reading from the CI_ENV variable was explicitly removed from the `index.php` file.  Was this due to not being able to get it to work?

I spent some time this morning trying to get this working as I run Cloudlog in a container, so being able to just set a variable, then change the variable if I need a different environment is significantly easier than trying to get on my host to edit the `index.php` file.  I run portainer as access to the backend of all my docker containers.

This is the line in `index.php` that is commented out: [Line 56](https://github.com/magicbug/Cloudlog/blob/19b6a7c62a5f1e9184cd3f01b3c050b0870a6676/index.php#L56)

I commented out Line 57 in my file, and uncommented 56.
I had a ton of issues where the environment variable wasn't being considered set originally.  I even tried to output the entire `$_Server` variable, and it was just blank.  If I called a bare `getenv()` I did see variables that were populated.

For any other users that prefer environment variables, this is the line that I finally got to work in my environment:
`define('ENVIRONMENT', getenv("CI_ENV") !== false ? getenv("CI_ENV") : 'development');`

**Comment by phl0** [comment]
I like the idea. Will add it to my installation. Would you see issues here to include this @magicbug?
